OffchainLabs/arbitrum-token-bridge
# refactor: get fetched chain pairs dynamically

Local networks are missing in the tx history chain pair list, so they are currently not being fetched.

We could hardcode them, but then we'd have to filter them out so that they do not show in prod.

It's better to use `getChains` as the source of truth because it has all the chains we need, regardless if we are connected to dev or prod.

Changed `multiChainFetchList` to a method in case of eventual race conditions with custom networks being added and then imported from the sdk.

### Steps to test
You can check manually in tx history to see if all chains are being fetched, or log all chains returned by `getMultiChainFetchList`.

Here I logged it myself and that's the results we get (mapped to network names, originally they are chain ids). It also includes one custom Orbit chain: Nitrogen.
<img width="402" alt="Screenshot 2024-01-25 at 16 02 51" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/47932951/1fe914c3-35f2-46f3-be44-ca34fcc5fefa">

Also renamed `parentChain` to `parentChainId`, and `chain` to `childChainId`.
